src/mlwm/build_inference_artifact.py

```diff
@@ -0,0 +1,350 @@
+"""
+Contents required when package to an inference artifact:
+
+- Model checkpoint, provided by a path to the CLI. Written to `checkpoint.pkl`
+- Extraction of statistics from training dataset. Paths to datasets inferred
+  from datastore configs link reference in neural-lam config provided to cli.
+  Written to `stats/{datastore_name}.stats.zarr`
+- Command line arguments that training was run with. Expected to be in
+  `training_cli_args.yaml`
+"""
+import os
+import shutil
+import sys
+import tempfile
+import zipfile
+from pathlib import Path
+
+import dotenv
+import mllam_data_prep as mdp
+import xarray as xr
+import yaml
+from loguru import logger
+from tqdm import tqdm
+from upath import UPath
+
+from . import __version__
+
+DEFAULT_ARTIFACT_PATH_FORMAT = (
+    "s3://mlwm-artifacts/inference-artifacts/{artifact_name}.zip"
+)
+
+dotenv.load_dotenv()
+
+ARTIFACT_PATH_FORMAT = os.environ.get(
+    "ARTIFACT_PATH_FORMAT", DEFAULT_ARTIFACT_PATH_FORMAT
+)
+
+TRAINING_CLI_ARGS_FILE = "training_cli_args.yaml"
+
+
+def _find_datastore_paths(nl_config_path: str):
```

Comment:
Maybe add a `-> dict:` just to make it easier/quicker to see what's returned.

Reply:
Yes, I will check for other missing type annotations

Reply:
done in https://github.com/dmidk/mlwm-deployment/pull/1/commits/ca3cd43a1a212ed79ad5b4c86f82fa37230e80ae